### Задание

Смоделировать игру против лотерейного автомата типа "777". Игрок платит 1 руб., после чего выпадает случайное целое число, равномерно распределенное от 0 до 999. При некоторых значениях числа игрок получает выигрыш.

Теоретически рассчитать средний выигрыш (проигрыш) и сравнить с результатами моделирования.

Вопросы:
- Выгодна ли игра игроку?
- Сколько в среднем приобретает или теряет игрок за одну игру?

In [1]:
import numpy as np

In [2]:
# Напишем функцию, которая проверяет число и возвращает сумму выигрыша:
def winning(number):
    win = {
        '777': 200,
        '999': 100,
        '555': 50,
        '333': 14,
        '111': 10,
        '77': 5,
        '7': 3,
        '00': 2,
        '0': 1
    }
    string_number = str(number)
    if string_number in win:
        return win[string_number]
    elif string_number[-2:] in win:
        return win[string_number[-2:]]
    elif string_number[-1] in win:
        return win[string_number[-1]]
    else:
        return 0  

In [32]:
attempt_price = 1
attempts = 1000000
all_results = np.random.randint(0, 1000, attempts)
prize_sum = 0

for res in all_results:
    prize_sum += winning(res)

print(f'Общий выигрыш: {prize_sum} руб.')
print(f'Среднее значение выигрыша: {prize_sum / (attempts * attempt_price)} руб.')
print(f'Итоговый результат игры: {prize_sum - attempt_price * attempts} руб.' )

Общий выигрыш: 800518 руб.
Среднее значение выигрыша: 0.800518 руб.
Итоговый результат игры: -199482 руб.


In [13]:
# Рассчитаем теоретический средний выигрыш:
x = 200 * (1 / 1000) + 100 * (1 / 1000) + 50 * (1 / 1000) \
    + 15 * (1 / 1000) + 10 * (1 / 1000) + 5 * (10 / 1000) \
    + 3 * (90 / 1000) + 2 * (9 / 1000) + 1 * (91 / 1000) 
print(x)

0.804


#### Выводы:

Теоретически рассчитанный средний выигрыш в данной игре составляет 0.804 рубля и соответствует полученным  результатам моделирования. В среднем, игрок будет проигрывать 20 копеек с каждой попытки стоимостью в 1 рубль, что делает игру невыгодной.